crates/pathfinder/src/rpc/v02/method/get_state_update.rs

```diff
@@ -0,0 +1,477 @@
+use anyhow::Context;
+
+use crate::{
+    core::BlockId,
+    rpc::v02::RpcContext,
+    storage::{StarknetBlocksBlockId, StarknetBlocksTable, StarknetStateUpdatesTable},
+};
+
+#[derive(serde::Deserialize, Debug, PartialEq, Eq)]
+pub struct GetStateUpdateInput {
+    block_id: BlockId,
+}
+
+crate::rpc::error::generate_rpc_error_subset!(GetStateUpdateError: BlockNotFound);
+
+#[allow(dead_code)]
+pub async fn get_state_update(
+    context: RpcContext,
+    input: GetStateUpdateInput,
+) -> Result<types::StateUpdate, GetStateUpdateError> {
+    let block_id = match input.block_id {
+        BlockId::Pending => {
+            let update = match &context.pending_data {
+                Some(pending) => pending.state_update().await,
+                None => None,
+            };
+            match update {
+                Some(update) => {
+                    let update = update.as_ref().clone().into();
+                    return Ok(update);
+                }
+                None => return Err(GetStateUpdateError::BlockNotFound),
+            }
+        }
+        BlockId::Latest => StarknetBlocksBlockId::Latest,
+        BlockId::Hash(hash) => hash.into(),
+        BlockId::Number(number) => number.into(),
+    };
+
+    let storage = context.storage.clone();
+    let span = tracing::Span::current();
+
+    let jh = tokio::task::spawn_blocking(move || {
+        let _g = span.enter();
+        let mut db = storage
+            .connection()
+            .context("Opening database connection")?;
+
+        let tx = db.transaction().context("Creating database transaction")?;
+
+        let block_hash = match block_id {
+            StarknetBlocksBlockId::Hash(h) => h,
+            StarknetBlocksBlockId::Number(_) | StarknetBlocksBlockId::Latest => {
+                StarknetBlocksTable::get_hash(
+                    &tx,
+                    block_id.try_into().expect("block_id is not a hash"),
+                )
+                .context("Read block from database")?
+                .ok_or(GetStateUpdateError::BlockNotFound)?
+            }
+        };
+
+        let state_update = StarknetStateUpdatesTable::get(&tx, block_hash)
+            .context("Read state update from database")?
+            .ok_or(GetStateUpdateError::BlockNotFound)?;
+
+        Ok(state_update.into())
+    });
+
+    jh.await.context("Database read panic or shutting down")?
+}
+
+mod types {
+    use serde::Serialize;
+    use serde_with::skip_serializing_none;
+
+    use std::collections::HashMap;
+
+    use crate::core::{
+        ClassHash, ContractAddress, ContractNonce, GlobalRoot, StarknetBlockHash, StorageAddress,
+        StorageValue,
+    };
+
+    #[skip_serializing_none]
+    #[derive(Clone, Debug, Serialize, PartialEq, Eq)]
+    #[serde(deny_unknown_fields)]
+    pub struct StateUpdate {
+        /// None for `pending`
+        #[serde(default)]
+        pub block_hash: Option<StarknetBlockHash>,
```

Comment:
Yes, it's not required according to the spec. We can't really do better than _not_ sending the block hash since that's missing for the pending state update...